File: Leet 4/solution.py
class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        sorted_nums: list[int] = sorted(nums1 + nums2)
        length: int = len(sorted_nums)

        if length % 2 == 0:
            at: int = length // 2
            return (sorted_nums[at] + sorted_nums[at - 1]) / 2

        return sorted_nums[length // 2]


def main():
    print('4. Median of Two Sorted Arrays')

    solution = Solution()
    case1 = solution.findMedianSortedArrays([1, 3], [2])
    print(case1)
    

if __name__ == '__main__':
    main()

File: Leet 4/solution.cs
// NOTE - left and right must be sorted
int[] MergeSortedLists(int[] nums1, int[] nums2)
{
    int[] result = new int[nums1.Length + nums2.Length];
    int nums1Idx = 0;
    int nums2Idx = 0;
    int idx = 0;

    while (nums1Idx < nums1.Length && nums2Idx < nums2.Length)
    {
        if (nums1[nums1Idx] < nums2[nums2Idx])
        {
            result[idx++] = nums1[nums1Idx++];
        }
        else
        {
            result[idx++] = nums2[nums2Idx++];
        }
    }

    while (nums1Idx < nums1.Length)
    {
        result[idx++] = nums1[nums1Idx++];
    }

    while (nums2Idx < nums2.Length)
    {
        result[idx++] = nums2[nums2Idx++];
    }

    return result;
}


float FindMedianSortedArrays(int[] nums1, int[] nums2)
{
    int[] merged = MergeSortedLists(nums1, nums2);
    int mid = merged.Length / 2;

    float result;
    if (merged.Length % 2 == 0)
    {
        result = (merged[mid] + merged[mid - 1]) / 2.0f;
    }
    else
    {
        result = merged[mid];
    }

    return result;
}


Console.WriteLine("4. Median of Two Sorted Arrays");

int[] nums1 = [1, 3];
int[] nums2 = [2];
Console.WriteLine($"Case1: {FindMedianSortedArrays(nums1, nums2)}");